### PR TITLE
Softmax: add weighted-sum normalization

### PR DESCRIPTION
**Pull request by madamczykhabana**
This introduces another algorithm for softmax normalization - wsum (weighted sum).
The main idea is that we can normalize softmax values by a weighted sum of the local block maximums, i.e. a = sum(w * max(attn)). In most cases the weights (or block_scales) are selected so be equal to 1/num_blocks so that the sum doesn't explode with a larger number of blocks.

### CODE REVIEW

File: vllm_hpu_extension/ops.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 import habana_frameworks.torch as htorch
 import torch
+import math
```
Comment (michalkuligowski) [inline vllm_hpu_extension/ops.py:11]: not needed

Comment (madamczykhabana) [reply vllm_hpu_extension/ops.py]: Thanks for catching that!

**Review by michalkuligowski — APPROVED** [review]
